File: lib/frontend/models/news_item.dart
class NewsItem {
  final String source;
  final String timeAgo;
  final String title;
  final String imageUrl;
  final String articleUrl;
  final List<String> tags;
  final int likes;
  final int comments;
  final String category;
  final DateTime publishDate;

  NewsItem({
    required this.source,
    required this.timeAgo,
    required this.title,
    required this.imageUrl,
    required this.articleUrl,
    required this.tags,
    required this.likes,
    required this.comments,
    this.category = 'Chung',
    DateTime? publishDate,
  }) : this.publishDate = publishDate ?? DateTime.now();
} 

File: vite-web/src/models/NewsItem.ts
export interface NewsItem {
  source: string;
  timeAgo: string;
  title: string;
  imageUrl: string;
  articleUrl: string;
  tags: string[];
  likes: number;
  comments: number;
  category: string;
  publishDate: string; // Will be converted to Date when used
} 